src/daemon/default_vm_image_vault.h

```diff
@@ -49,7 +50,8 @@ class DefaultVMImageVault final : public BaseVMImageVault
     ~DefaultVMImageVault();
 
     VMImage fetch_image(const FetchType& fetch_type, const Query& query, const PrepareAction& prepare,
-                        const ProgressMonitor& monitor) override;
+                        const ProgressMonitor& monitor, const bool unlock,
+                        const std::optional<std::string> checksum) override;
```

Comment:
I think this should be a `const` reference.

Reply:
Do you mean the `checksum` argument? I thought `optional` was implemented as a pointer so the refetence was not needed. But anyway, it makes the function signature look neat.

Reply:
Yes, that argument.  Thanks for changing this.  From https://en.cppreference.com/w/cpp/utility/optional, it says:

> If an optional<T> contains a value, the value is guaranteed to be allocated as part of the optional object footprint, i.e. no dynamic memory allocation ever takes place. Thus, an optional object models an object, not a pointer, even though [operator*()](https://en.cppreference.com/w/cpp/utility/optional/operator*) and [operator->()](https://en.cppreference.com/w/cpp/utility/optional/operator*) are defined.

Reply:
Amazing, thanks!